packages/bot-web-ui/src/stores/save-modal-store.ts

```diff
@@ -75,9 +77,63 @@ export default class SaveModalStore implements ISaveModalStore {
 
             this.setButtonStatus(button_status.COMPLETED);
         }
-
-        this.updateBotName(bot_name);
-        saveWorkspaceToRecent(xml, is_local ? save_types.LOCAL : save_types.GOOGLE_DRIVE);
+        const {
+            dashboard: { active_tab },
+        } = this.root_store;
+        const {
+            load_modals: { selected_strategy_id, setRecentStrategies },
+        } = this.root_store;
+
+        if (active_tab === 0) {
+            const workspace_id = selected_strategy_id || Blockly.utils.genUid();
+            const workspace = await getSavedWorkspaces();
+            const current_workspace_index = workspace.findIndex(strategy => strategy.id === workspace_id);
+            let save_type = save_types.UNSAVED;
+
+            if (is_local) {
+                save_type = save_types.LOCAL;
+            } else {
+                save_type = save_types.GOOGLE_DRIVE;
+            }
+            if (save_as_collection) {
+                save_type = save_types.UNSAVED;
+            }
+
+            if (current_workspace_index >= 0) {
+                const current_workspace = {
+                    id: workspace_id,
+                    xml: Blockly.Xml.domToText(xml),
+                    name: bot_name,
+                    timestamp: Date.now(),
+                    save_type,
+                };
+                workspace[current_workspace_index] = current_workspace;
+            } else {
+                workspace.push({
+                    id: workspace_id,
+                    timestamp: Date.now(),
+                    name: bot_name,
+                    xml: Blockly.Xml.domToText(xml),
+                    save_type,
```

Comment:
the structure is exactly as we have above ... so maybe we can assign it once and use it in if condition like: 
const workspace_structure = {
                    id: workspace_id,
                    timestamp: Date.now(),
                    name: bot_name,
                    xml: Blockly.Xml.domToText(xml),
                    save_type,
}